### ISSUE TITLE
[8.0.0+] minSpareRows serious performance downgrade

### ISSUE THREAD

**Issue by AMBudnik** [handsontable/handsontable#7410]
### Description

After getting a report from one of our users I've tested a scenario
 ```js
    data: Handsontable.helper.createSpreadsheetData(1000, 1000),
    rowHeaders: true,
    colHeaders: true,
    minSpareRows: 5000
```
using Handsontable v.7.4.2 and 8.0.0+

Here's the result
![Zrzut ekranu 2020-11-23 o 11 28 15](https://user-images.githubusercontent.com/10757813/99951838-000ed280-2d7f-11eb-9025-9a978ebc7386.png)

### Your environment
* Handsontable version: 8.0.0
* Browser Name and version: Chrome 87
* Operating System: macOS Catalina 10.15.7

Inform 25724

**Comment by AMBudnik** [comment]
I've also added v8.2.0 that looks a bit better than 8.0.0, but still worse than 7.4.2

![Zrzut ekranu 2020-11-23 o 11 36 20](https://user-images.githubusercontent.com/10757813/99952616-3436c300-2d80-11eb-92a2-cf14c27179f8.png)

**Comment by wojciechczerniak** [comment]
`minSpareRows: 5000`? 😮  `minSpareRows` is creating new rows. You have 5000 calls to `createRow`:
https://github.com/handsontable/handsontable/blob/34793d5e9ee10176941ebe2589996daa80b5effd/src/core.js#L596-L607

Two ideas though:

1. `suspendCalculations` that @budnix is evaluating in #7274 may help here. We should not recalculate IndexMapper each time a row is added. ~But this may be already batched in updateSettings.~ Nope: `adjustRowsAndCols` is outside of  `initIndexMappers` (and therefore outside of `batch`) so we may have a room for improvement: https://github.com/handsontable/handsontable/blob/34793d5e9ee10176941ebe2589996daa80b5effd/src/core.js#L2063 https://github.com/handsontable/handsontable/blob/34793d5e9ee10176941ebe2589996daa80b5effd/src/core.js#L1971
2. It's possible that we may optimize it by not calling `datamap.createRow` multiple times. The `datamap.createRow` accepts the number of rows to add and it would be done once vs 5k calls.
3. We were going to rewrite `minSpareRows` as virtual rows that are not part of the dataset in `v8.x` branch but that was put on hold by the new roadmap.

TL;DR; I see this as expected behavior of `v8.0.0`. The change we've made with IndexMappers was supposed to invert the logic and CRUD operations like `createRow` will take longer while `render` is lighter and faster. Such a number of `minSpareRows` is not what we're optimizing for.

----
Same will happen for `minSpareCols`
https://github.com/handsontable/handsontable/blob/34793d5e9ee10176941ebe2589996daa80b5effd/src/core.js#L624-L632

**Comment by AMBudnik** [comment]
Thank you for the input @wojciechczerniak

**Comment by aninde** [comment]
@AMBudnik don't close it right now, I need to make proper performance investigation.

**Comment by wojciechczerniak** [comment]
Done in #7439 and released in v8.3.0. If you need to test it more track it on your personal list
